src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,76 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import { useState } from 'react';
+import { Status } from '../../types/Status';
+
+type TodoFilterProps = {
+  onFilter: (status: Status) => void,
+  onTitleFilter: (title: string) => void,
+  onClearFilter: () => void,
+};
+
+export const TodoFilter
+= ({ onFilter, onTitleFilter, onClearFilter } : TodoFilterProps) => {
+  const [query, setQuery] = useState('');
+  const [filter, setFilter] = useState(Status.all);
```

Comment:
btw, why do you need the separate state for the filter here, if you can just pass the filter from the app.tsx as the prop